agent/eventhandler/task_handler_types.go

```diff
@@ -71,16 +72,27 @@ func (event *sendableEvent) taskArn() string {
 func (event *sendableEvent) taskShouldBeSent() bool {
 	event.lock.RLock()
 	defer event.lock.RUnlock()
+
 	if event.isContainerEvent {
 		return false
 	}
 	tevent := event.taskChange
 	if tevent.Status == api.TaskStatusNone {
 		return false // defensive programming :)
 	}
-	if event.taskSent || (tevent.Task != nil && tevent.Task.GetSentStatus() >= tevent.Status) {
+	if event.taskSent {
 		return false // redundant event
 	}
+	if tevent.Task != nil && tevent.Task.GetSentStatus() >= tevent.Status {
+		for _, containerStateChange := range tevent.Containers {
+			container := containerStateChange.Container
```

Comment:
Please add some comments here to makes it easier to read. And this part should have a unit test to cover.